File: src/schemas/supplier.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class SupplierBase(BaseModel):
    name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None
    country_id: int
    region_id: int
    area_id: int
    longitude: Optional[float] = None
    latitude: Optional[float] = None
    user_id: int
    is_active: Optional[bool] = True


class SupplierCreate(SupplierBase):
    pass


class SupplierUpdate(SupplierBase):
    pass


class SupplierOut(SupplierBase):
    id: int
    created_at: datetime

    class Config:
        orm_mode = True

File: src/Hackathon.Domain/Entities/Supplier.cs
using InhaTestBot.Domain.Commons;

namespace InhaTestBot.Domain.Entities;

public class Supplier : Auditable
{
    public string Name { get; set; }
    public string Email { get; set; }
    public string Phone { get; set; }
    public string Address { get; set; }

    public long CountryId { get; set; }
    public Country Country { get; set; }

    public long RegionId { get; set; }
    public Region Region { get; set; }

    public long AreaId { get; set; }
    public Area Area { get; set; }

    public decimal? Longitude { get; set; }
    public decimal? Latitude { get; set; }

    public long UserId { get; set; }
    public User User { get; set; }

    public bool IsActive { get; set; }
}
